### PR TITLE
Fix #6269: Move Onboarding into a separate SPM target

### LINKED ISSUE #6269: Move Onboarding into a separate SPM target
## Problem Description 

Onboarding related classes, views , assets should be moved to separate SPM target. cDependencies should be taken care of accordingly.

This will include classes for Welcome Screen and Callouts.

### PR DESCRIPTION
**Pull request by soner-yuksel**
## Summary of Changes

This pull request fixes #6269

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:

Sanity check for Onboarding is required

## Screenshots: 

Screenshot 2022-10-27 at 4 55 18 PM

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Package.swift
Comment (iccub) [inline Package.swift]: This file is in two places now?

File: Client/Frontend/Browser/BrowserViewController.swift
Comment (iccub) [inline Client/Frontend/Browser/BrowserViewController.swift]: Didn't this file move to the `BraveOnboarding`?

File: Package.swift
Comment (kylehickinson) [inline Package.swift]: Can we name this just `Onboarding` or `AppOnboarding` something? It's not a Brave product like VPN/News/Wallet/Talk is.

File: Sources/BraveOnboarding/OnboardingPreferences.swift
Comment (kylehickinson) [inline Sources/BraveOnboarding/OnboardingPreferences.swift]: This may cause ambiguity since `ClientPreferences` already defines it.  We can create a new name specifically for this module i.e. `Preferences.Onboarding`.  The keys for the containing options can remain the same

**Review by iccub — APPROVED** [review]
Seems to work fine for me now.
Let's wait for @kylehickinson final review before we merge

**Review by kylehickinson — CHANGES_REQUESTED** [review]
Just a few nits

One day it would be good to also refactor a lot of the logic that happens in BVC into this new module around onboarding. At the moment this just moves code over to a new folder but adds `public` to basically everything. We can eventually move to expose less and less

**Comment by soner-yuksel** [comment]
> Just a few nits
> 
> One day it would be good to also refactor a lot of the logic that happens in BVC into this new module around onboarding. At the moment this just moves code over to a new folder but adds `public` to basically everything. We can eventually move to expose less and less

Totally agreed. There is a very tightly knitted logic there inside bvc extension and such but now with this move it will be easier for me to uncouple the logic more and make some nice changes.

File: Sources/BraveOnboarding/OnboardingPreferences.swift
Comment (soner-yuksel) [reply Sources/BraveOnboarding/OnboardingPreferences.swift]: Fixed at https://github.com/brave/brave-ios/pull/6270/commits/1b5bbd88bd88e295e76cecf0c75da4fe09932d17

File: Package.swift
Comment (soner-yuksel) [reply Package.swift]: Fixed at https://github.com/brave/brave-ios/pull/6270/commits/1b5bbd88bd88e295e76cecf0c75da4fe09932d17

**Review by kylehickinson — APPROVED** [review]

File: Client/Frontend/Browser/BrowserViewController.swift
Comment (soner-yuksel) [reply Client/Frontend/Browser/BrowserViewController.swift]: Fixed https://github.com/brave/brave-ios/pull/6270/commits/1b831115560e4c68a2751ac5f53a1fa0361716f1

Great Catch, thank you!

File: Package.swift
Comment (soner-yuksel) [reply Package.swift]: Fixed at https://github.com/brave/brave-ios/pull/6270/commits/1b831115560e4c68a2751ac5f53a1fa0361716f1

**Pull request merged**
